sources/payment-hub/src/main/java/hu/dpc/rt/psp/routebuilder/gsma/channel/MerchantPayment.java

```diff
@@ -0,0 +1,59 @@
+package hu.dpc.rt.psp.routebuilder.gsma.channel;
+
+import hu.dpc.rt.psp.config.*;
+import hu.dpc.rt.psp.dto.gsma.TransactionObject;
+import org.apache.camel.CamelContext;
+import org.apache.camel.builder.RouteBuilder;
+import org.apache.camel.model.dataformat.JsonLibrary;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.HttpMethod;
+
+@Configuration
+public class MerchantPayment extends RouteBuilder {
+
+    /*
+     * What should be the routes? First, someone makes a payment through their local app - that is done through
+     * Fineract in our case. That request is received by the hub and forwarded to the server.
+     * What also needs to happen is checking if the transaction is completed successfully or not.
+     */
```

Comment:
Have you assumed asynchronous txn posting from Fineract followed by txn status polling?